### PR TITLE
Implement dependencies on required font tables in OpenType format

### PR DESCRIPTION
**Pull request by brendanzab**
This allows for dependencies between the required font tables in OpenType. I still need to figure out how to allow for dependencies on optional font tables, but I think it’s nice to make progress on this!

To do this I added the following builtins to Fathom:

- `Option : Type -> Type`
- `some : fun (A : Type) -> A -> Option A`
- `none : fun (A : Type) -> Option A`
- `fold :  fun (A : Type) (B : Type) -> B -> (A -> B) -> Option A -> B`
- `array8_find : fun (len : U8) (A : Type) -> (A -> Bool) -> Array8 len A -> Option A`
- `array16_find : fun (len : U16) (A : Type) -> (A -> Bool) -> Array16 len A -> Option A`
- `array32_find : fun (len : U32) (A : Type) -> (A -> Bool) -> Array32 len A -> Option A`
- `array64_find : fun (len : U64) (A : Type) -> (A -> Bool) -> Array64 len A -> Option A`
- `unwrap : fun (A : Type) -> Option A -> Format` (where `Repr (unwrap A option_a) = A`)

### CODE REVIEW

**Comment by mikeday** [comment]
Very exciting stuff!

File: doc/reference.md [hunk 1]
```diff
@@ -416,14 +420,27 @@ parsing.
...
 The fail format should never produce a term, so is represented with [void](#void).
 
 | format | `Repr` format |
 | ------ | ------------- |
 | `fail` | `Void`        |
 
+### Unwrap format
+
+An unwrap format succeeds with the data contained in a the `some` case of an
+[option](#options), or fails.
```
[the remark this answers is not in the document]
Comment (brendanzab) [reply doc/reference.md:434]: Oh, yes, that is a good point!

**Comment by wezm** [comment]
All those new built-ins look like they will be super useful!

**Comment by brendanzab** [comment]
I might go ahead and merge this if folks are cool with it! Unless you want to merge #351 first?

**Comment by wezm** [comment]
Fine with me, merge away!

**Pull request merged**